### PR TITLE
feat/archive previous menus

### PR DESCRIPTION
**Pull request by kamilfedio**
- added pivot table for meals and hashes_website with additional info such as price and size (moved from meals) to easier track changes
- added endpoint for getting whole history

### CODE REVIEW

**Review by czaja307 — COMMENTED** [review]
pretty solid work, i've left some suggestions as comments in code snippets, as always there is always room for discussion

also i commited fix because you left some unused imports, which messes up with build on production, but i think it was just a little oversight

File: app/controllers/meals_controller.ts [hunk 1]
```diff
@@ -1,16 +1,83 @@
...
+        todayMeals = await getMealsByHash(secondLastHash.hash)
+      }
+
+      const meals = todayMeals.map((singleMeal) => ({
+        ...singleMeal.meal.serialize(),
+        price: singleMeal.price,
+        size: singleMeal.size,
+      }))
+
+      return response.status(200).json({ meals, isMenuOnline: isMenuOnline })
```
Comment (simon-the-shark) [inline app/controllers/meals_controller.ts:37]: `{ meals, isMenuOnline: isMenuOnline }` is the same as `{ meals, isMenuOnline }`

File: app/models/meal.ts [hunk 2]
```diff
@@ -21,15 +23,12 @@ export default class Meal extends BaseModel {
...
-  declare price: number
-
   @column.dateTime({ autoCreate: true })
   declare createdAt: DateTime
 
   @column.dateTime({ autoCreate: true, autoUpdate: true })
   declare updatedAt: DateTime
+
+  @hasMany(() => HashesMeal)
+  declare hashes: relations.HasMany<typeof HashesMeal>
```
[the remark this answers is not in the document]
Comment (simon-the-shark) [reply app/models/meal.ts:33]: mostly matter of preference but it's good to be at least consistent and in the previous file, you didn't use this namespace import

**Comment by kamilfedio** [comment]
done

File: app/controllers/meals_controller.ts (hunk 1 above)
Comment (czaja307) [reply app/controllers/meals_controller.ts]: you can't run from the watchful eye of the mr. Shark

**Review by czaja307 — APPROVED** [review]
LGTM

super robota Kamil! 🔥🚀

**Review by simon-the-shark — APPROVED** [review]
if it works then it works. nice job guys

**Pull request merged**